vertx-config/src/main/java/io/vertx/config/impl/spi/EnvVariablesConfigStore.java

```diff
@@ -20,30 +20,40 @@
  * @author <a href="http://escoffier.me">Clement Escoffier</a>
  */
 public class EnvVariablesConfigStore implements ConfigStoreFactory, ConfigStore {
+  private final boolean jsonConversion;
+
   private JsonObject cached;
 
+  public EnvVariablesConfigStore() {
+    this(true);
+  }
+
+  public EnvVariablesConfigStore(boolean jsonConversion) {
+    this.jsonConversion = jsonConversion;
+  }
+
   @Override
   public String name() {
     return "env";
   }
 
   @Override
   public ConfigStore create(Vertx vertx, JsonObject configuration) {
-    return this;
+    return new EnvVariablesConfigStore(configuration.getBoolean("json-conversion", false));
```

Comment:
@cescoffier that is indeed my prefered naming, but since you had suggested `json-conversion` I chose to stick with that. I will amend this _asap_.